code/datums/status_effects/status_effect.dm

```diff
@@ -112,17 +112,17 @@
 	. = FALSE
 	var/datum/status_effect/S1 = effect
 	LAZYINITLIST(status_effects)
+	var/list/arguments = args.Copy()
+	arguments[1] = src
 	for(var/datum/status_effect/S in status_effects)
 		if(S.id == initial(S1.id) && S.status_type)
 			if(S.status_type == STATUS_EFFECT_REPLACE)
 				S.be_replaced()
 			else if(S.status_type == STATUS_EFFECT_REFRESH)
-				S.refresh()
+				S.refresh(arglist(arguments))
```

Comment:
i cant tell why youre doing this, or why youre passing src as an argument since it doesnt seem to be used

Reply:
If refresh() doesn't have the same arguments as the proc to create a new status effect than it doesn't function as intended. For example, if you dump ants onto someone and then dump another beaker of ants onto them after, without this the debuff would be reset entirely, instead of adding more ants onto the person. 
As for the src bit, I was told to move that up by lemon
![image](https://user-images.githubusercontent.com/66052067/125865253-c3f06fc4-9e62-479a-8313-fde3c2588d43.png)